### PR TITLE
Migrate some OptionalInt's to Optional<Integer>s

### PR DESCRIPTION
**Pull request by RatanRSur**
This change is not intended to change any functionality.

### CODE REVIEW

File: besu/src/main/java/org/hyperledger/besu/Runner.java [hunk 1]
```diff
@@ -215,14 +215,16 @@ private void writeBesuPortsToFile() {
...
+                enode
+                    .getDiscoveryPort()
+                    .ifPresent(
+                        discoveryPort ->
+                            properties.setProperty("discovery", String.valueOf(discoveryPort)));
+                enode
+                    .getDiscoveryPort()
+                    .ifPresent(
+                        listeningPort ->
+                            properties.setProperty("p2p", String.valueOf(listeningPort)));
```
Comment (davemec) [inline besu/src/main/java/org/hyperledger/besu/Runner.java:227]: Is this change just to make it more functional style? I'd be careful doing that as it requires testing and doesn't actually improve anything.

Comment (RatanRSur) [reply besu/src/main/java/org/hyperledger/besu/Runner.java:227]: In this case the problem is `OptionalInt`. It's a non-generic functional-ish interface that lacks a lot of the safety features of `Optional`s so you end up having to using unsafe methods like `getAsInt`. Like I've mentioned before, using a Maybe monad is not just a matter of style. It allows the compiler to force you to handle the non-existence case explicitly, thus reducing bugs in the long run. Here's an example of a purely style change:
```
var result = 0
for (int i =0; i < arrList.size(); ++i) {
  if (i %2 ==0) result +=i
}
```
vs
```
arrList.stream().filter(i -> i %2 ==0).reduce(0, (accumulator, element) -> accumulator + element)
```
There's no additional type information here that the second has that the first doesn't have.

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgent.java [hunk 2]
```diff
@@ -187,13 +186,7 @@ protected void handleIncomingPacket(final Endpoint sourceEndpoint, final Packet
...
             .getPacketData(PingPacketData.class)
             .flatMap(PingPacketData::getFrom)
-            .flatMap(
-                fromEndpoint -> {
-                  final OptionalInt maybePort = fromEndpoint.getTcpPort();
-                  return maybePort.isPresent()
-                      ? Optional.of(maybePort.getAsInt())
-                      : Optional.empty();
-                })
+            .flatMap(Endpoint::getTcpPort)
```
Comment (shemnon) [inline ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgent.java:189]: This is a concise argument as to why OptionalInt should be avoided.

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/peers/EnodeURL.java [hunk 3]
```diff
@@ -40,22 +41,18 @@
...
+      final Optional<Integer> discoveryPort) {
     checkArgument(
         nodeId.size() == NODE_ID_SIZE, "Invalid node id.  Expected id of length: 64 bytes.");
-    checkArgument(
-        !listeningPort.isPresent() || NetworkUtility.isValidPort(listeningPort.getAsInt()),
-        "Invalid listening port.  Port should be between 1 - 65535.");
-    checkArgument(
-        !discoveryPort.isPresent() || NetworkUtility.isValidPort(discoveryPort.getAsInt()),
-        "Invalid discovery port.  Port should be between 1 - 65535.");
+    listeningPort.ifPresent(port -> NetworkUtility.checkPort(port, "listening"));
```
Comment (shemnon) [inline ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/peers/EnodeURL.java:54]: Will exceptions be thrown out of this lambda construct?  We should have a unit test to validate, all the unit tests I reviewed are all within UInt16.

Comment (RatanRSur) [reply ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/peers/EnodeURL.java:54]: Is it not higher here?

https://github.com/RatanRSur/besu/blob/5720fdcbcfb887e4255de37add482f0c3ef4fa0b/ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/peers/EnodeURLTest.java#L276

Comment (shemnon) [reply ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/peers/EnodeURL.java:54]: I was scanning for the builder variants (`.discoveryPort(int)` and `.listeningPort(int)`.)  Are there negative port tests?  We should test those too.  Possibly via the builders.

**Pull request merged**